test/controllers/users_controller_test.rb

```diff
@@ -2,4 +2,59 @@
 
 describe UsersController do
 
+  describe "Logged in users" do
+
+    before do
```

Comment:
This `before` block is causing that CSRF error/warning.  Instead just put `perform_login(users(:grace))` inside each `it` block.